File: src/main/java/yj/capstone/aerofarm/domain/Device.java
package yj.capstone.aerofarm.domain;

import javax.persistence.*;

@Entity
public class Device extends BaseEntity{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nickname;

    private int temperature;

    private int humidity;

    /**
     * 양액의 경우 ml 단위로 할건지?
     * float 고려
     */
    private int fertilizer;

    private int brightness;

    private String deviceImage;

    @Enumerated(EnumType.STRING)
    private Model model;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "plant_id")
    private Plant plant;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id")
    private Member owner;
}

File: lib/model/Device.dart
class Device{
  String nickname="";

  String imageUrl="";

  //기기 고유의 MAC주소
  String macAddress="";

  String ipAddress="";

  //CD키 개념, 변경 가능
  String uuid="";

  String model="";

  String plant="";

  String owner="";
  String createDate = "";

  int number=0;

  int temperature = 20;
  int humidity = 50;
  int fertilizer = 1;
  bool ledOn = false;

  Device(Map<String,dynamic> data){
    if(data['plant']!=null) {
      plant = data['plant'];
    }else{
      plant = "미등록";
    }
    nickname = data['nickname'];
    imageUrl = data['imageUrl'];
    model = data['model'];
    createDate = data['createDate'].toString();
  }
}